File: demo-app/app/src/javaApp/java/com/launchkey/android/authenticator/demo/ui/adapter/DemoFeatureAdapter.java
package com.launchkey.android.authenticator.demo.ui.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.launchkey.android.authenticator.demo.R;


public class DemoFeatureAdapter extends BaseAdapter {

    private final @NonNull Context mContext;
    private final @NonNull int[] mItems;

    public DemoFeatureAdapter(Context context, int[] i) {
        mContext = context;
        mItems = i;
    }

    @Override
    public int getCount() {
        return mItems.length;
    }

    @Override
    public Integer getItem(int position) {
        return mItems[position];
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        TextView v = (TextView) convertView;

        if (v == null) {
            v = (TextView) LayoutInflater.from(mContext)
                    .inflate(R.layout.demo_activity_list_item, parent, false);
        }

        v.setText(mContext.getString(getItem(position)));

        return v;
    }
}

File: demo-app/app/src/kotlinApp/java/com/launchkey/android/authenticator/demo/ui/adapter/DemoFeatureAdapter.kt
package com.launchkey.android.authenticator.demo.ui.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import com.launchkey.android.authenticator.demo.R

class DemoFeatureAdapter(private val mContext: Context, private val mItems: IntArray) : BaseAdapter() {
    override fun getCount(): Int {
        return mItems.size
    }

    override fun getItem(position: Int): Int {
        return mItems[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        var v = convertView as TextView?
        if (v == null) {
            v = LayoutInflater.from(mContext)
                    .inflate(R.layout.demo_activity_list_item, parent, false) as TextView
        }
        v.text = mContext.getString(getItem(position))
        return v
    }

}